### DISCUSSION TITLE
Include hyperlink within a dict response

### DISCUSSION THREAD

**Discussion by witk44** [fastapi/fastapi#8342] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/")
async def root():
     hyperlink_format = '<a href="http://freshpeople/">Fresh</a>'
     res = {"Name": "test", "link": hyperlink_format}
     return res
```

### Description

I am trying to create a list of dicts with all employees of a company with their names being hyperlinks to their own individual routes conatining all of their employee information. I can return hyperlink_format by itself with response_class=HTMLResponse however I am trying to conatin this hyperlink within a dict.

### Operating System

Windows

### FastAPI Version

0.78.0

### Python Version

Python 3.10.5

**Comment by zoliknemet** [comment]
I am not sure what is the question. Do you want for response to be HTML or JSON.
I did small example on how i see it right now. 
**BIG NOTE**: if you are making REST API avoid rendering HTML on backend side - that should frontend side do

```python
import uvicorn as uvicorn

from fastapi import FastAPI
from typing import List
from pydantic import BaseModel


app = FastAPI()


class User(BaseModel):
    id: int
    full_name: str


users = [
    User(id=1, full_name="John Smit"),
    User(id=2, full_name="Jane Smit"),
    User(id=3, full_name="Foo"),
    User(id=4, full_name="Bar"),
]


class UserResponse(User):
    link: str


@app.get("/users", response_model=List[UserResponse])
async def root():
    users_response = []
    for user in users:
        users_response.append(
            {
                "id": user.id,
                "full_name": user.full_name,
                "link": f"<a href='http://freshpeople/{user.id}'>{user.full_name}</a>",
            }
        )

    return users_response


if __name__ == "__main__":
    uvicorn.run(app)


```

Response
```json
[
  {
    "id": 1,
    "full_name": "John Smit",
    "link": "<a href='http://freshpeople/1'>John Smit</a>"
  },
  {
    "id": 2,
    "full_name": "Jane Smit",
    "link": "<a href='http://freshpeople/2'>Jane Smit</a>"
  },
  {
    "id": 3,
    "full_name": "Foo",
    "link": "<a href='http://freshpeople/3'>Foo</a>"
  },
  {
    "id": 4,
    "full_name": "Bar",
    "link": "<a href='http://freshpeople/4'>Bar</a>"
  }
]
```
I don't know if this helps you

**Comment by witk44** [comment]
The class and list of users is what i want to do but its missing the hyperlink. Essentially I don't want a link in the dict but instead it will be {"Name": Foo, "id":2} but when Foo is clicked it redirects me to 'http://freshpeople/2'. In html you can do this by 
 href='http://freshpeople/2'> Foo and if i return this (a href='http://freshpeople/2'>Foo) with response_class=HTMLResponse it works as a hyperlink but i can't retunr the list of dicts as well just the hyperlinks, so i want to combine the response you provided with what i just described. If at all possible, I hope that cleared up my question as well.

**Comment by zoliknemet** [comment, marked as answer]
I get it now, you would like to return HTML and JSON at the same time.
There is no way of doing that.
HTTP does not work that way. HTTP has header `Content-Type` which has only 1 value so you have to define of what type is your response.
If you are returning HTML it is set as `text/html` and if you are returning JSON it is set to `application/json`

I think that your overall design is not right considering best practices and conventions.
